### PR TITLE
fix(Irrelevant contact events deletion):

### PR DESCRIPTION
**Pull request by daniel7764**
-Now warning message appears only on clinical details save
-Handled the case when getInvestigatedDates returns undefined

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import IsolationSource from 'models/IsolationSource';
 import DBAddress, {initDBAddress} from 'models/DBAddress';
 import useCustomSwal from 'commons/CustomSwal/useCustomSwal';
 import { setFormState } from 'redux/Form/formActionCreators';
+import useSymptomsFields from './SymptomsFields/useSymptomsFields';
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:13]: imports order

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 2]
```diff
@@ -231,60 +233,71 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
...
             const earliestDateToInvestigate = getDatesToInvestigate(doesHaveSymptoms, symptomsStartDate, coronaTestDate)[0];
-            deleteIrrelevantEventsLogger.info('Sending to server date to delete contact events by', Severity.LOW);
-            axios.delete('/intersections/deleteContactEventsByDate', {params: {earliestDateToInvestigate}}).then((result) => {
-                if(result.data?.data?.deleteContactEventsBeforeDate) {
-                    deleteIrrelevantEventsLogger.info('Deleting contact events finished with success', Severity.LOW);
-                } else {
-                    deleteIrrelevantEventsLogger.error(`Deleting contact events finished with errors: ${result.data}`, Severity.LOW);
+            if(Boolean(earliestDateToInvestigate)) {
+                deleteIrrelevantEventsLogger.info('Sending to server date to delete contact events by', Severity.LOW);
+                axios.delete('/intersections/deleteContactEventsByDate', {params: {earliestDateToInvestigate}}).then((result) => {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:238]: what about setIsLoading?

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 3]
```diff
@@ -231,60 +233,71 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
         });
         if(Boolean(coronaTestDate)) {
             const earliestDateToInvestigate = getDatesToInvestigate(doesHaveSymptoms, symptomsStartDate, coronaTestDate)[0];
-            deleteIrrelevantEventsLogger.info('Sending to server date to delete contact events by', Severity.LOW);
-            axios.delete('/intersections/deleteContactEventsByDate', {params: {earliestDateToInvestigate}}).then((result) => {
-                if(result.data?.data?.deleteContactEventsBeforeDate) {
-                    deleteIrrelevantEventsLogger.info('Deleting contact events finished with success', Severity.LOW);
-                } else {
-                    deleteIrrelevantEventsLogger.error(`Deleting contact events finished with errors: ${result.data}`, Severity.LOW);
+            if(Boolean(earliestDateToInvestigate)) {
```
Comment (OrelKaravani) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:236]: Check the array length first and then extract the "[0]" element inside the if condition.

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 1 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:13]: where? seems good to me

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 3 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts]: fixed

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 2 above)
Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts]: fixed

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts (hunk 1 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:13]: it should be in the third section

Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts]: oh got it, fixed

**Review by ItayBenMoshe — APPROVED** [review]

**Review by OrelKaravani — APPROVED** [review]

**Pull request merged**